src/wasm/wat-lexer.cpp

```diff
@@ -943,7 +943,7 @@ void Lexer::lexToken() {
 }
 
 TextPos Lexer::position(const char* c) const {
-  assert(size_t(c - buffer.data()) < buffer.size());
+  assert(size_t(c - buffer.data()) <= buffer.size());
```

Comment:
If the buffer is of size 1 then `&buffer[1] - buffer.data()` is equal to 1, so this check passes, but that's out of bounds isn't it? Or is there a padding byte..?

Reply:
There's no padding byte, but `c` is never dereferenced, so that's ok.

Reply:
I see now, thanks.